stl/inc/functional

```diff
@@ -1479,6 +1479,69 @@ namespace placeholders {
     _INLINE_VAR constexpr _Ph<20> _20{};
 } // namespace placeholders
 
+#if _HAS_CXX20
+// FUNCTION TEMPLATE _Call_front_binder
+template <size_t... _Ix, class _Cv_FD, class _Cv_tuple_TiD, class... _Unbound>
+auto _Call_front_binder(index_sequence<_Ix...>, _Cv_FD&& _Obj, _Cv_tuple_TiD&& _Tpl, _Unbound&&... _Unbargs) {
+    return _STD invoke(_STD forward<_Cv_FD>(_Obj), _STD get<_Ix>(_STD forward<_Cv_tuple_TiD>(_Tpl))...,
+        _STD forward<_Unbound>(_Unbargs)...);
+}
+
+// CLASS TEMPLATE _Front_binder
+template <class _Fx, class... _Types>
+class _Front_binder { // wrap bound callable object and arguments
+private:
+    static_assert(is_constructible_v<decay_t<_Fx>, _Fx>, "bind_front requires a constructible decayed callable");
+    static_assert(is_move_constructible_v<decay_t<_Fx>>, "bind_front requires a move constructible decayed callable");
+    static_assert(conjunction_v<is_constructible<decay_t<_Types>, _Types>...>,
+        "bind_front requires constructible decayed bound arguments");
+    static_assert(conjunction_v<is_move_constructible<decay_t<_Types>>...>,
+        "bind_front requires move constructible decayed bound arguments");
+
+    using _Seq    = index_sequence_for<_Types...>;
+    using _First  = decay_t<_Fx>;
+    using _Second = tuple<decay_t<_Types>...>;
+
+    _Compressed_pair<_First, _Second> _Mypair;
+
+public:
+    explicit _Front_binder(_Fx&& _Func, _Types&&... _Args)
+        : _Mypair(_One_then_variadic_args_t(), _STD forward<_Fx>(_Func), _STD forward<_Types>(_Args)...) {}
+
+    template <class... _Unbound>
+        auto operator()(_Unbound&&... _Unbargs)
+        & noexcept(is_nothrow_invocable_v<decay_t<_Fx>&, decay_t<_Types>&..., _Unbound...>) {
+        return _Call_front_binder(_Seq(), _Mypair._Get_first(), _Mypair._Myval2, _STD forward<_Unbound>(_Unbargs)...);

```

Comment:
Our style is inconsistent, but in new code we try to use `{}` to construct tag types (so it looks less like a function call), so this should be `_Seq{}`. Occurs below.